### PR TITLE
EVG-12942 more logging for failed task archive

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: model/lifecycle.go [hunk 1]
```diff
@@ -389,6 +389,23 @@ func RestartVersion(versionId string, taskIds []string, abortInProgress bool, ca
...
+				for _, dbTask := range finishedTasks {
+					dbTasks = append(dbTasks, idAndExecution{Id: dbTask.Id, Execution: dbTask.Execution})
+				}
+				grip.Debug(message.WrapError(err, message.Fields{
+					"ticket":          "EVG-12942",
+					"message":         "error archiving task",
+					"task_id":         t.Id,
+					"execution":       t.Execution,
+					"display_only":    t.DisplayOnly,
+					"query_resuts":    dbTasks,
```
Comment (ablack12) [inline model/lifecycle.go:406]: typo

File: model/lifecycle.go [hunk 2]
```diff
@@ -389,6 +389,23 @@ func RestartVersion(versionId string, taskIds []string, abortInProgress bool, ca
 	for _, t := range finishedTasks {
 		if !t.IsPartOfSingleHostTaskGroup() { // for single host task groups we don't archive until fully restarting
 			if err = t.Archive(); err != nil {
+				type idAndExecution struct {
+					Id        string `json:"id"`
+					Execution int    `json:"execution"`
+				}
+				dbTasks := []idAndExecution{}
+				for _, dbTask := range finishedTasks {
```
Comment (ablack12) [inline model/lifecycle.go:397]: archive destructively modifies execution; would it be bad for performance to get this information before ever going into the loop? if so, then maybe just include results for up to the failing task, or else the data in this list is inconsistent and that might be confusing

**Review by ablack12 — CHANGES_REQUESTED** [review]

File: model/lifecycle.go (hunk 2 above)
Comment (john-m-liu) [reply model/lifecycle.go:397]: I changed the field name so it's a bit clearer but I think it's fine

File: model/lifecycle.go (hunk 1 above)
Comment (john-m-liu) [reply model/lifecycle.go:406]: oops fixed

File: model/lifecycle.go (hunk 2 above)
Comment (ablack12) [reply model/lifecycle.go:397]: Well I was talking about dbTasks: if there are 5 tasks,  and we've successfully archived task one and two, then task 3 fails, task 1 and 2 will show the updated execution  and task 3, 4, and 5 won't. But that itself could be revealing so.

**Review by ablack12 — APPROVED** [review]

File: model/lifecycle.go (hunk 2 above)
Comment (john-m-liu) [reply model/lifecycle.go:397]: Yep that was my intent. It'll definitely be confusing if someone seeing the log line hasn't read the code carefully, but I think that's a reasonable expectation here

**Pull request merged**